src/commands/publish.rs

```diff
@@ -57,73 +80,57 @@ fn warn_site_incompatible_route(deploy_config: &DeployConfig) {
+            return Ok(namespace); // Sites binding already exists; ignore
         }
     }
 
-    target.add_kv_namespace(KvNamespace {
+    let site_namespace = KvNamespace {
         binding: "__STATIC_CONTENT".to_string(),
         id: site_namespace.id,
-        bucket: Some(site_config.bucket.to_owned()),
-    });
-    Ok(())
+    };
+
+    target.add_kv_namespace(site_namespace.clone());
+
+    Ok(site_namespace)
 }
 
-pub fn upload_buckets(
-    target: &Target,
-    user: &GlobalUser,
-    verbose: bool,
-) -> Result<Option<AssetManifest>, failure::Error> {
-    let mut asset_manifest = None;
-    for namespace in &target.kv_namespaces() {
-        if let Some(bucket) = &namespace.bucket {
-            // We don't want folks setting their bucket to the top level directory,
-            // which is where wrangler commands are always called from.
-            let current_dir = env::current_dir()?;
-            if bucket.as_os_str() == current_dir {
-                failure::bail!(
-                    "{} You need to specify a bucket directory in your wrangler.toml",
-                    emoji::WARN
-                )
-            }
-            let path = Path::new(&bucket);
-            if !path.exists() {
-                failure::bail!(
-                    "{} bucket directory \"{}\" does not exist",
-                    emoji::WARN,
-                    path.display()
-                )
-            } else if !path.is_dir() {
-                failure::bail!(
-                    "{} bucket \"{}\" is not a directory",
-                    emoji::WARN,
-                    path.display()
-                )
-            }
-            let manifest_result = kv::bucket::sync(target, user, &namespace.id, path, verbose)?;
-            if target.site.is_some() {
-                if asset_manifest.is_none() {
-                    asset_manifest = Some(manifest_result)
-                } else {
-                    // only site manifest should be returned
-                    unreachable!()
-                }
-            }
-        }
+// We don't want folks setting their bucket to the top level directory,
+// which is where wrangler commands are always called from.
+pub fn assert_non_root_bucket(bucket: &PathBuf) -> Result<(), failure::Error> {
+    // TODO: this should really use a convenience function for "Wrangler Project Root"
+    let current_dir = env::current_dir()?;
+    if bucket.as_os_str() == current_dir {
+        failure::bail!(
+            "{} You need to specify a bucket directory in your wrangler.toml",
+            emoji::WARN
+        )
+    }
+    let path = Path::new(&bucket);
+    if !path.exists() {
+        failure::bail!(
+            "{} bucket directory \"{}\" does not exist",
+            emoji::WARN,
+            path.display()
+        )
+    } else if !path.is_dir() {
+        failure::bail!(
+            "{} bucket \"{}\" is not a directory",
+            emoji::WARN,
+            path.display()
+        )
```

Comment:
these failures are not related to if it is a root bucket, should probably rename the function 

(worth noting that it's also invalid to have bucket be `dist` for webpack types but we can leave that for another PR)